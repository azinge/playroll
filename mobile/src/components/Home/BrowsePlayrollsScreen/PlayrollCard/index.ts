import PlayrollCard from './PlayrollCard';
export default PlayrollCard;
