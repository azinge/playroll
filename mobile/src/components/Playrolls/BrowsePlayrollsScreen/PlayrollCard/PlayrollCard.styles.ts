import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  submitButtonLeft: {
    // flex: 1,
    // position: 'absolute',
    // left: 0,
    // bottom: 0,
    width: '50%',

    backgroundColor: 'purple',
    padding: 5,
    borderBottomLeftRadius: 12,
    borderRightWidth: 0.25,
    borderRightColor: 'white',
  },
  submitButtonRight: {
    // flex: 1,
    // position: 'absolute',
    // left: 0,
    // bottom: 0,
    width: '50%',

    backgroundColor: 'purple',
    padding: 5,
    borderBottomRightRadius: 12,
    borderLeftWidth: 0.25,
    borderLeftColor: 'white',
  },
  submitButtonText: {
    color: 'white',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: 18,
  },
});
