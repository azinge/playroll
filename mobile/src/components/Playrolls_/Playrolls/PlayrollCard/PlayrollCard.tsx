/**
 * PlayrollCard
 */

import * as React from "react";
import { Text, View, Dimensions, Image, Button } from "react-native";
import Carousel from "react-native-snap-carousel";
import { Playroll, Roll, MusicSource } from "../../../graphql/types";
import { Card } from "react-native-elements";

export interface Props {
  playroll: Playroll;
  editPlayroll: () => void;
}

interface State {}

export default class PlayrollCard extends React.Component<Props, State> {
  _renderItem({ item, index }: { item: Roll; index: number }) {
    let source: MusicSource | null = null;
    if (item.data && item.data.sources && item.data.sources.length > 0) {
      source = item.data.sources[0];
    }
    return (
      <View style={{ height: 100 }} key={index}>
        {source && (
          <Image
            source={{ uri: source.cover }}
            style={{ height: 100, width: 100, borderRadius: 5 }}
          />
        )}
      </View>
    );
  }
  render() {
    const { playroll, editPlayroll } = this.props;
    return (
      <Card
        title={playroll.name}
        // image={require("../../assets/wack.jpg")}
        key={playroll.id}
      >
        <View style={{ flexDirection: "row" }}>
          <View style={{ height: 110 }}>
            {playroll.rolls.length > 0 ? (
              <Carousel
                data={playroll.rolls}
                renderItem={this._renderItem}
                hasParallaxImages={false}
                sliderWidth={110}
                itemWidth={100}
                itemHeight={100}
                layout={"tinder"}
                layoutCardOffset={5}
              />
            ) : (
              <Image
                source={{
                  uri:
                    "https://www.unesale.com/ProductImages/Large/notfound.png",
                }}
                style={{ height: 110, width: 110, borderRadius: 5 }}
              />
            )}
          </View>
          <Button onPress={editPlayroll} title="Edit Playroll" />
        </View>
      </Card>
    );
  }
}
