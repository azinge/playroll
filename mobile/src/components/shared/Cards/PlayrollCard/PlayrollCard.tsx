/**
 * PlayrollCard
 */

import * as React from 'react';
import {
  View,
  Image,
  Button,
  TouchableOpacity,
  ActivityIndicator,
  Text,
  FlatList,
  TouchableHighlight,
} from 'react-native';
import Carousel from 'react-native-snap-carousel';

import {
  DeletePlayrollMutation,
  GetCurrentUserPlayrollQuery,
} from '../../../../graphql/requests/Playroll/';

import { LIST_CURRENT_USER_PLAYROLLS } from '../../../../graphql/requests/Playroll/ListCurrentUserPlayrollsQuery';

import { Playroll, Roll, MusicSource } from '../../../../graphql/types';
import styles from './PlayrollCard.styles';
import Heading from '../../Text/Heading';
import HorizontalRule from '../../Text/HorizontalRule';

export interface Props {
  playroll?: Playroll;
  onPress?: (playroll) => void;
  hideCreator?: boolean;
}

interface State {}

export default class PlayrollCard extends React.Component<Props, State> {
  _renderItem({ item, index }: { item: Roll; index: number }) {
    let source: MusicSource | null = null;
    if (item.data && item.data.sources && item.data.sources.length > 0) {
      source = item.data.sources[0];
    }
    return (
      <View style={{ height: 100 }} key={index}>
        {source && (
          <Image
            source={{ uri: source.cover }}
            style={{ height: 75, width: 75, borderRadius: 5, marginRight: 5 }}
          />
        )}
      </View>
    );
  }

  render() {
    const { onPress = () => {} } = this.props;
    console.log(this.props.playroll.id);
    return (
      <GetCurrentUserPlayrollQuery variables={{ id: this.props.playroll.id }}>
        {({ loading, error, data }) => {
          const playroll: any =
            (data && data.private.currentUserPlayroll) || {};
          let creator = '';
          if (playroll.user && playroll.user.name) {
            if (playroll.user.accountType === 'Managed') {
              creator = 'Playroll';
            } else {
              creator = playroll.user.name;
            }
          }
          return (
            <View
              onLayout={event => {
                let { x, y, width, height } = event.nativeEvent.layout;
              }}
            >
              <TouchableHighlight
                style={{
                  marginHorizontal: 15,
                  marginVertical: 15,
                  height: 75,
                }}
                onPress={() => onPress(playroll)}
              >
                <View
                  style={{
                    flexDirection: 'row',
                    alignSelf: 'center',
                    justifyContent: 'center',
                    alignContent: 'center',
                    alignItems: 'center',
                  }}
                >
                  <View
                    style={{
                      alignSelf: 'center',
                      justifyContent: 'center',
                      alignContent: 'center',
                      alignItems: 'center',
                    }}
                  >
                    {playroll.rolls && playroll.rolls.length > 0 ? (
                      <Carousel
                        data={playroll.rolls}
                        renderItem={this._renderItem}
                        hasParallaxImages={false}
                        sliderWidth={100}
                        itemWidth={75}
                        itemHeight={75}
                        loop={true}
                      />
                    ) : (
                      <Image
                        source={{
                          uri:
                            'https://www.unesale.com/ProductImages/Large/notfound.png',
                        }}
                        style={{ height: 75, width: 75, borderRadius: 5 }}
                      />
                    )}
                  </View>
                  <View style={{ flex: 1, alignItems: 'center' }}>
                    <Heading type={'h6'} alignment={'left'}>
                      {playroll.name}
                    </Heading>
                    {!this.props.hideCreator && (
                      <Heading type={'h9'} alignment={'left'} color={'grey'}>
                        By {creator}
                      </Heading>
                    )}
                  </View>
                </View>
              </TouchableHighlight>
              <HorizontalRule />
            </View>
          );
        }}
      </GetCurrentUserPlayrollQuery>
    );
  }
}
