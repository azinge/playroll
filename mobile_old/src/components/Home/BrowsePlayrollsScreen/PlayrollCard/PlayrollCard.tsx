/**
 * PlayrollCard
 */

import * as React from "react";
import { Text, View, Dimensions, Image, Button } from "react-native";
import Carousel from "react-native-snap-carousel";
import { Card } from "react-native-elements";
import { NavigationScreenProp } from "react-navigation";

import {
  DeletePlayrollMutation,
  GetPlayrollQuery,
} from "../../../../graphql/requests/Playroll/";

import { LIST_CURRENT_USER_PLAYROLLS } from "../../../../graphql/requests/Playroll/ListCurrentUserPlayrollsQuery";

import { Playroll, Roll, MusicSource } from "../../../../graphql/types";

export interface Props {
  playroll?: Playroll;
  editPlayroll?: () => void;
  navigation?: NavigationScreenProp<{}>;
}

interface State {}

export default class PlayrollCard extends React.Component<Props, State> {
  _renderItem({ item, index }: { item: Roll; index: number }) {
    let source: MusicSource | null = null;
    if (item.data && item.data.sources && item.data.sources.length > 0) {
      source = item.data.sources[0];
    }
    return (
      <View style={{ height: 100 }} key={index}>
        {source && (
          <Image
            source={{ uri: source.cover }}
            style={{ height: 100, width: 100, borderRadius: 5 }}
          />
        )}
      </View>
    );
  }
  render() {
    const { playroll = {}, editPlayroll = () => {} } = this.props;
    return (
      <GetPlayrollQuery
        variables={{ id: playroll.id }}
        fetchPolicy="cache-only"
      >
        {({ data }) => {
          const playroll = (data && data.playroll) || {};
          return (
            <Card
              title={playroll.name}
              // image={require("../../assets/wack.jpg")}
              key={playroll.id}
            >
              <View style={{ flexDirection: "row" }}>
                <View style={{ height: 110 }}>
                  {playroll.rolls && playroll.rolls.length > 0 ? (
                    <Carousel
                      data={playroll.rolls}
                      renderItem={this._renderItem}
                      hasParallaxImages={false}
                      sliderWidth={110}
                      itemWidth={100}
                      itemHeight={100}
                      layout={"tinder"}
                      layoutCardOffset={5}
                      loop={true}
                    />
                  ) : (
                    <Image
                      source={{
                        uri:
                          "https://www.unesale.com/ProductImages/Large/notfound.png",
                      }}
                      style={{ height: 110, width: 110, borderRadius: 5 }}
                    />
                  )}
                </View>
                <View>
                  <Button onPress={editPlayroll} title="Edit Playroll" />
                  <DeletePlayrollMutation
                    variables={{
                      id: playroll.id,
                    }}
                    refetchQueries={[LIST_CURRENT_USER_PLAYROLLS]}
                  >
                    {(deletePlayroll, { data }) => (
                      <Button
                        title="Delete Playroll"
                        onPress={() => {
                          deletePlayroll();
                        }}
                      />
                    )}
                  </DeletePlayrollMutation>
                </View>
              </View>
            </Card>
          );
        }}
      </GetPlayrollQuery>
    );
  }
}
