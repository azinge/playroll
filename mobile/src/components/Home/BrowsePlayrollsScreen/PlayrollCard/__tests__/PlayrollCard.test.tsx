import * as React from "react";
import renderer from "react-test-renderer";
import { MockedProvider } from "react-apollo/test-utils";

import PlayrollCard from "../PlayrollCard";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MockedProvider>
        <PlayrollCard />
      </MockedProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
