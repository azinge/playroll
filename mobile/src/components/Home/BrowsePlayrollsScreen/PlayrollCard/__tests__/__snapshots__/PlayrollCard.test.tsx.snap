// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "height": 110,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://www.unesale.com/ProductImages/Large/notfound.png",
            }
          }
          style={
            Object {
              "borderRadius": 5,
              "height": 110,
              "width": 110,
            }
          }
        />
      </View>
      <View>
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Edit Playroll
            </Text>
          </View>
        </View>
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Delete Playroll
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
